src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -221,12 +221,18 @@ void setupSchema(List<ColumnMetadata> columns) {
    * synchronization
    */
   void close() {
-    if (this.vectorsRoot != null) {
```

Comment:
Can you say a little more about what's happening here?  When would we want to clear the fields objects but not the  vectors and allocators?  If the worry is that we close the buffer before flushing it, should we put checks for that case specifically instead of having buffers in a weird state where they have open allocators but near to be reinitialized because their fields are empty?

Reply:
the fields object is just a map and we will clean it all the time. For vectors and allocators, we want to clean them up if the channel is invalid because we will skip flushing it so the buffer may have leftover data. If there is no issue, there will always be a flush before calling this close, so the vectors and allocators should already be empty, so we want to call allocator.close directly which will verify if there any memory leak or not, does that make more sense now?

Reply:
I think I understand, but I don't see why you'd want to call `Utils.closeAllocator()` in one case and `this.allocator.close()` in another case.  Wouldn't you want to close child allocators in both cases?

Reply:
There is no child allocator for in this case, `Utils.closeAllocator()` is used in other places which we do have child allocators if you check the usage

Reply:
Why wouldn't we have child allocators in this case?  Also the `Utils.closeAllocator()` handles the case of having no child allocators.  Are you trying to avoid calling `alloc.releaseBytes(alloc.getAllocatedMemory());`?

Reply:
> Why wouldn't we have child allocators in this case?

Because we don't create it, please take a look at how the child allocators are created in `SnowflakeStreamingIngestChannelInternal` and passed down to `ArrowRowBuffer`

> Are you trying to avoid calling alloc.releaseBytes(alloc.getAllocatedMemory());?

Yes, that's another reason, because if we release it first, we won't be able to verify that there is no memory leak